frontend/src/metabase/setup/components/EmbeddingSetup/steps/TableSelectionStep.tsx

```diff
@@ -257,3 +260,29 @@ const retry = async <T,>({
   }
   onMaxTries?.();
 };
+
+function useShouldFetchTables(setLoading: (loading: boolean) => void) {
```

Comment:
This fixes [EMB-485: Make the "Connect your data" wait for the db sync](https://linear.app/metabase/issue/EMB-485/make-the-connect-your-data-wait-for-the-db-sync)

This uses the same approach as the database sync status component currently.
https://github.com/metabase/metabase/blob/a442965526ae77a83b3ca36667f17ba8e1d047e6/frontend/src/metabase/status/containers/DatabaseStatus/DatabaseStatus.tsx#L16-L17